File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {

        int n = strs.size();
        string res = "";

        if (n == 0) {
            return "";
        }

        sort(strs.begin(), strs.end());

        string first = strs[0];
        string last = strs[n - 1];

        for (int i = 0; i < min(first.size(), last.size()); i++) {

            if (first[i] != last[i]) {
                return res;
            }
            res += first[i];
        }
        return res;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    # 3. Longest Common Prefix (LeetCode 14)
    def longestCommonPrefix(self, strs: list[str]) -> str:
        # Handle edge case where array is empty
        if not strs:
            return ""
        
        # Assume the first string is the prefix and shrink it progressively
        prefix = strs[0]
        for s in strs[1:]:
            # While the current string does not start with the prefix, shorten the prefix
            while s.find(prefix) != 0:
                prefix = prefix[:-1]
                if not prefix:
                    return ""
        return prefix